### ISSUE TITLE
Logs are misleading when transaction is sampled due to parent being sampled

### ISSUE THREAD

**Issue by rmsy** [getsentry/sentry-ruby#1712]
### Issue Description

Hello 👋

If a transaction has a parent which is sampled, the tracing debug logs are slightly misleading. Due to the precedence of the conditionals below, the `sample_rate` may be `false` or `0.0` if a sampler proc is defined and returns one of them:
https://github.com/getsentry/sentry-ruby/blob/8538717cfcaf2f1a1f71817eb06a4f85de91205d/sentry-ruby/lib/sentry/transaction.rb#L108-L115

This results in the following log message being logged, indicating the transaction will be discarded:
https://github.com/getsentry/sentry-ruby/blob/8538717cfcaf2f1a1f71817eb06a4f85de91205d/sentry-ruby/lib/sentry/transaction.rb#L125-L129

However, if `@parent_sampled` is true for the transaction, it is still sent in `finish`:
https://github.com/getsentry/sentry-ruby/blob/8538717cfcaf2f1a1f71817eb06a4f85de91205d/sentry-ruby/lib/sentry/transaction.rb#L167-L173

I assume the first two conditionals in the `sample_rate` assignment can just be reversed, but I wasn't 100% sure if doing that would have any other side-effects, so I opened this issue instead of a PR.

### Reproduction Steps

1. Configure Sentry to have a `traces_sampler` proc which returns `false` or `0.0`
2. Send a request to Rails with a `Sentry-Trace` header
3. Observe the debug log message indicating the transaction will be discarded
4. Observe that the transaction is still sent to Sentry

### Expected Behavior

The log message should not indicate that the transaction will be discarded.

### Actual Behavior

The log message indicates that the transaction will be discarded.

### Ruby Version

2.7.4

### SDK Version

4.8.2

### Integration and Its Version

Rails

**Comment by st0012** [comment]
After digging into this a bit, I think this there's a conflict between 2 tracing requirements or my understanding on them.

1. We should prioritize `traces_sampler`'s decision over inherited decision. This follows [sentry-python's implementation](https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/tracing.py#L654-L656).
2. Always send the transaction if its parent is sampled, which is the `@parent_sampled ` in this line

https://github.com/getsentry/sentry-ruby/blob/8538717cfcaf2f1a1f71817eb06a4f85de91205d/sentry-ruby/lib/sentry/transaction.rb#L167

I think 2 is probably wrong as `sentry-python` doesn't implement this and it kind of disobeys the decision of `set_initial_sample_decision`. So removing the `@parent_sampled` part should be the right thing to do.

@sl0thentr0py Can you help me check if the above is true?

**Comment by sl0thentr0py** [comment]
@st0012, the official precedence order requirement is written down [here](https://develop.sentry.dev/sdk/performance/#sampling), so I believe yes, 1. is correct - `traces_sampler` takes priority.

**Comment by st0012** [comment]
@sl0thentr0py sorry that I wasn't clear about what "the above" meant. I meant that my 2nd point was incorrect and the condition should be changed.

**Comment by sl0thentr0py** [comment]
yes, 2. is incorrect and the condition should be changed.. The transaction should be sent based on the sampling decision in 1. where `traces_sampler` takes priority over the parent.

**Comment by st0012** [comment]
@sl0thentr0py thanks for the confirmation!

@rmsy in this case, the logging logic is fine. it's the transaction should not be sent, which I'll fix shortly.

if you want to keep the current behavior (send unsampled transaction based on its parent's decision), you need prioritize the inherited sampling decision in your `traces_sampler`:

```rb
config.traces_sampler = lambda do |sampling_context|
  parent_sampled = sampling_context[:parent_sampled]

  if !parent_sampled.nil?
    parent_sampled
  else
    # the rest of sampling logic
  end
end
```
